Remove redundant placeholders in forms docs

tojimjiang opened the pull request:
Fixes #28260

tojimjiang committed:
Remove redundant placeholders in forms docs

Resolves #28260

patrickhlauke commented on site/docs/4.3/components/forms.md:
```diff
@@ -23,7 +23,7 @@ Here's a quick example to demonstrate Bootstrap's form styles. Keep reading for
   <div class="form-group">
     <label for="exampleInputPassword1">Password</label>
-    <input type="password" class="form-control" id="exampleInputPassword1" placeholder="Password">
+    <input type="password" class="form-control" id="exampleInputPassword1" placeholder="Enter password">
```
i'd remove this placeholder altogether, as well as the `placeholder="Enter email"` in the previous input

patrickhlauke commented on site/docs/4.3/components/forms.md:
```diff
@@ -799,21 +799,21 @@ Custom feedback styles apply custom colors, borders, focus styles, and backgroun
     <div class="col-md-3 mb-3">
       <label for="validationCustom04">State</label>
-      <input type="text" class="form-control" id="validationCustom04" placeholder="State" required>
+      <input type="text" class="form-control" id="validationCustom04" placeholder="Choose..." required>
```
"choose..." as placeholder for a text input doesn't make sense. for consistency with the previous form earlier in the page, suggest replacing this with an actual `<select>` instead

patrickhlauke commented on site/docs/4.3/components/forms.md:
```diff
@@ -867,34 +867,34 @@ While these feedback styles cannot be styled with CSS, you can still customize t
     <div class="col-md-3 mb-3">
       <label for="validationDefault04">State</label>
-      <input type="text" class="form-control" id="validationDefault04" placeholder="State" required>
+      <input type="text" class="form-control" id="validationDefault04" placeholder="Choose..." required>
```
"choose..." as placeholder for a text input doesn't make sense. for consistency with the previous form earlier in the page, suggest replacing this with an actual `<select>` instead

patrickhlauke commented on site/docs/4.3/components/forms.md:
```diff
@@ -947,21 +947,21 @@ We recommend using client-side validation, but in case you require server-side v
     <div class="col-md-3 mb-3">
       <label for="validationServer04">State</label>
-      <input type="text" class="form-control is-invalid" id="validationServer04" placeholder="State" required>
+      <input type="text" class="form-control is-invalid" id="validationServer04" placeholder="Choose..." required>
```
"choose..." as placeholder for a text input doesn't make sense. for consistency with the previous form earlier in the page, suggest replacing this with an actual `<select>` instead

patrickhlauke commented on site/docs/4.3/components/forms.md:
```diff
@@ -1075,21 +1075,21 @@ If your form layout allows it, you can swap the `.{valid|invalid}-feedback` clas
     <div class="col-md-3 mb-3">
       <label for="validationTooltip04">State</label>
-      <input type="text" class="form-control" id="validationTooltip04" placeholder="State" required>
+      <input type="text" class="form-control" id="validationTooltip04" placeholder="Choose..." required>
```
"choose..." as placeholder for a text input doesn't make sense. for consistency with the previous form earlier in the page, suggest replacing this with an actual `<select>` instead

tojimjiang replied:
These 4 have been changed to `<select>`.

tojimjiang committed:
Remove more placeholders

Fix context for states
Remove redundant commands

tojimjiang committed:
Merge branch 'master' of https://github.com/twbs/bootstrap

tojimjiang committed:
Change state field to select

Include "Choose..." placeholder for select
Fix typo .form-select is not a Bootstrap class

tojimjiang committed:
Resolve linting errors

patrickhlauke commented on site/docs/4.3/components/forms.md:
```diff
@@ -807,7 +807,7 @@ Custom feedback styles apply custom colors, borders, focus styles, and backgroun
     <div class="col-md-3 mb-3">
       <label for="validationCustom04">State</label>
       <select id="inputState" class="form-control" id="validationCustom04" required>
```
per the description just before the example/code block, best to also change the class to `class="custom-select"`

patrickhlauke replied:
remove the redundant `id="inputState"`

patrickhlauke commented on site/docs/4.3/components/forms.md:
```diff
@@ -961,7 +961,7 @@ We recommend using client-side validation, but in case you require server-side v
     <div class="col-md-3 mb-3">
       <label for="validationServer04">State</label>
       <select class="form-control is-invalid" id="validationServer04" required>
```
again, change `class="form-control"` to `class="custom-select"`

patrickhlauke commented on site/docs/4.3/components/forms.md:
```diff
@@ -1092,7 +1092,7 @@ If your form layout allows it, you can swap the `.{valid|invalid}-feedback` clas
     <div class="col-md-3 mb-3">
       <label for="validationTooltip04">State</label>
       <select class="form-control" id="validationTooltip04" required>
```
and again `class="custom-select"`

patrickhlauke reviewed (changes requested):
sorry, last few minor changes. then it looks good to go :+1:

mdo commented on site/docs/4.3/components/forms.md:
```diff
@@ -799,21 +799,24 @@ Custom feedback styles apply custom colors, borders, focus styles, and backgroun
       <label for="validationCustom04">State</label>
-      <input type="text" class="form-control" id="validationCustom04" placeholder="State" required>
+      <select id="inputState" class="form-control" id="validationCustom04" required>
+        <option selected disabled value = ''>Choose...</option>
```
```suggestion
        <option selected disabled value='">Choose...</option>
```

mdo commented on site/docs/4.3/components/forms.md:
```diff
@@ -867,34 +870,37 @@ While these feedback styles cannot be styled with CSS, you can still customize t
       <label for="validationDefault04">State</label>
-      <input type="text" class="form-control" id="validationDefault04" placeholder="State" required>
+      <select class="form-control" id="validationDefault04" required>
+        <option selected disabled value = ''>Choose...</option>
```
```suggestion
        <option selected disabled value="">Choose...</option>
```

mdo commented on site/docs/4.3/components/forms.md:
```diff
@@ -947,21 +953,24 @@ We recommend using client-side validation, but in case you require server-side v
       <label for="validationServer04">State</label>
-      <input type="text" class="form-control is-invalid" id="validationServer04" placeholder="State" required>
+      <select class="form-control is-invalid" id="validationServer04" required>
+        <option selected disabled value = ''>Choose...</option>
```
```suggestion
        <option selected disabled value="">Choose...</option>
```

mdo commented on site/docs/4.3/components/forms.md:
```diff
@@ -1075,21 +1084,24 @@ If your form layout allows it, you can swap the `.{valid|invalid}-feedback` clas
       <label for="validationTooltip04">State</label>
-      <input type="text" class="form-control" id="validationTooltip04" placeholder="State" required>
+      <select class="form-control" id="validationTooltip04" required>
+        <option selected disabled value = ''>Choose...</option>
```
```suggestion
        <option selected disabled value="">Choose...</option>
```

mdo reviewed (commented):
Please be sure to follow code standards used everywhere in this project—no spaces around attribute's `=` and double quotes around attribute values. Fixed these here with suggested changes.

tojimjiang committed:
Change validated select to .custom-select

Update validated select attributes to follow code standards

tojimjiang commented:
I have updated the validated selects to use `.custom-select`, and attribute values to follow the code standards.

MartijnCuppens committed:
Merge branch 'master' into master

XhmikosR committed:
Merge branch 'master' into master

tojimjiang committed:
Merge branch 'master' into master

XhmikosR commented:
@patrickhlauke: should we backport this to v4?

patrickhlauke commented:
> should we backport this to v4?

yes

XhmikosR commented:
OK, I'll cherry pick this to the new v4-dev branch.

XhmikosR committed:
Merge branch 'master' into master
